### PR TITLE
Added skeletonLocker

### PR DESCRIPTION
**Pull request by vvasco**
This PR introduces `skeletonLocker`, which is an extra-layer applied on the top of `skeletonRetriever` to enforce spatio-temporal consistency on a skeleton identified by its tag. The tag can be specified through the provided thrift services. 

Each skeleton has two pivots (`shoulder_center` and `hip_center`) associated to a sphere whose radius increases over time if no association is found, up to a maximum of `max-radius`. 
`skeletonLocker` adds an additional skeleton to the `opc`, appending to the original tag the key `-locked`. Each skeleton in the `opc` is compared to the locked one, and the closer one falling into the sphere is associated to it.

The following shows a typical example where a skeleton gets lost due to the drag effect given by the movement of the robot base: 
- using only `skeletonRetriever`, the skeleton `#0` gets lost and a skeleton with a new tag is created;
- coupling `skeletonRetriever` and `skeletonLocked`, the skeleton `#0-locked` is added to the `opc` and is tracked for the whole dataset.

| Not locked (only `skeletonRetriever`) | Locked (`skeletonRetriever` + `skeletonLocker`) |
| ------------- | ------------- |  
| <p align="center">  </p> |   <p align="center">  </p> |

### CODE REVIEW

**Comment by pattacini** [comment]
Superb @vvasco 👍 
Before I'll have a look into it, could you please detail here a quick summary of the algorithm behind?

**Comment by pattacini** [comment]
In the meantime, can you also review the PR in order to address the issues raised by Codacy?

**Comment by vvasco** [comment]
Once the tag of the skeleton to lock is set, `skeletonLocker` starts tracking it, adding a new skeleton to the `opc`. 
This skeleton has two pivots associated with a sphere with an initial radius defined by `radius` param.
If the original skeleton is lost, for each new skeleton found in the `opc`, `skeletonLocker` computes the distance between corresponding 3D pivots, selecting the minimum distance: 
- if the distance is less than the sphere radius, the new skeleton is associated to the locked skeleton (if several skeletons are found, the closest one is associated). The locked skeleton and its pivots are updated consequently;
- otherwise, to increase the probability of finding a skeleton to associate to the locked one, the radius increases with an exponential law given by `radius*exp(dt/time-trend)`, with `dt` being the time elapsed since the last successful association, up to a maximum radius defined by `max-radius`.

File: modules/skeletonLocker/src/main.cpp [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+ ******************************************************************************/
+
+/**
+ * @file main.cpp
+ * @authors: Valentina Vasco <valentina.vasco@iit.it>
+ */
+
+#include <limits>
+#include <algorithm>
+#include <math.h>
```
Comment (pattacini) [inline modules/skeletonLocker/src/main.cpp]:
```suggestion
#include <cmath>
```

File: modules/skeletonLocker/src/main.cpp [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+                    t=Time::now();
+                    opcSet(*locked);
+                }
+            }
+            cout<<"Skeletons: "<<endl;
+            for (int i=0; i<new_skeletons.size(); i++)
+            {
+                cout<<new_skeletons[i].skeleton->getTag()<<" score: "<<new_skeletons[i].score<<" locked skeleton radius: "<<locked->radius<<endl;
+            }
+            cout<<endl;
```
Comment (pattacini) [inline modules/skeletonLocker/src/main.cpp]: Don't use `cout`; better off relying on `yInfo` instead.

**Comment by pattacini** [comment]
Documentation updated as well 👍

**Pull request merged**